docs/dev/system-design/integrated-code-lifecycle/integrated-code-lifecycle-system-design.inc

```diff
@@ -95,6 +95,22 @@ The service provides the functionality for an Artemis user to interact with buil
 The user can access this functionality using the UI over a set of endpoints provided by a REST API.
 The ``LocalCIResultProcessingService`` retrieves the build job results which were generated by the build agents from the result queue. It is responsible for grading the build job results, notifying the user and persisting information on the build job execution in the database.
 
+.. _ci_priority:
+
+Priority Calculation
+""""""""""""""""""""
+Build jobs are assigned a priority value when they are added to the job queue. This value is used to determine the order in which build jobs are executed.
+The higher the priority value, the sooner the build job is executed.
+
+The priority value is determined based on several factors related to the programming exercise and its context. If the build is part of a batch, for example if an instructors triggers a re-build of all submissions of an exercise, it is assigned the lowest priority of ``4``.
+For exam exercises, those that are part of a test exam or a test run within an exam are assigned a priority of ``2``.
+Submissions made after the due date, such as those in practice mode or finished exams, receive a lower priority of ``3``.
+Exercises that are part of an ongoing exam are given a higher priority of ``1`` to ensure quick feedback. Optional exercises, which are not included in the overall score, are assigned a priority of ``3``.
+Additionally, if the exercise is part of a test course, a penalty is added to the priority, making it less urgent. The penalty is implemented by adding a value of ``5`` to the priority.
+The priority calculation is implemented in the ``LocalCITriggerService``.
```

Comment:
In general, the new documentation is comprehensive. I think it would be a nice touch to order the different priorities by their level (from 5 to 1 or vice versa).